packages/browser-agent-core/features/ajax/aggregate/index.js

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { registerHandler as register } from '../../../common/event-emitter/register-handler'
+import { stringify } from '../../../common/util/stringify'
+import { nullable, numeric, getAddStringContext, addCustomAttributes } from '../../../common/serialize/bel-serializer'
+import { handle } from '../../../common/event-emitter/handle'
+import { getConfigurationValue, getInfo } from '../../../common/config/config'
+import { HarvestScheduler } from '../../../common/harvest/harvest-scheduler'
+import { subscribeToUnload } from '../../../common/unload/unload'
+import { setDenyList, shouldCollectEvent } from '../../../common/deny-list/deny-list'
+import { FeatureBase } from '../../../common/util/feature-base'
+
+export class Aggregate extends FeatureBase {
+  constructor(agentIdentifier, aggregator, externalFeatures = []) {
+    super(agentIdentifier, aggregator, externalFeatures)
+    let ajaxEvents = []
+    let spaAjaxEvents = {}
+    let sentAjaxEvents = []
+    let scheduler
+
+    const ee = this.ee
+    const externalHarvestKeys = ['jserrors']
+
+    const harvestTimeSeconds = getConfigurationValue(agentIdentifier, 'ajax.harvestTimeSeconds') || 10
+    const MAX_PAYLOAD_SIZE = getConfigurationValue(agentIdentifier, 'ajax.maxPayloadSize') || 1000000
+
+    // Exposes these methods to browser test files -- future TO DO: can be removed once these fns are extracted from the constructor into class func
+    this.storeXhr = storeXhr;
+    this.prepareHarvest = prepareHarvest;
+    this.getStoredEvents = function() { return { ajaxEvents, spaAjaxEvents } };
+
+    ee.on('interactionSaved', (interaction) => {
+      if (!spaAjaxEvents[interaction.id]) return
+      // remove from the spaAjaxEvents buffer, and let spa harvest it
+      delete spaAjaxEvents[interaction.id]
+    })
+
+    ee.on('interactionDiscarded', (interaction) => {
+      if (!spaAjaxEvents[interaction.id] || !allAjaxIsEnabled()) return
+
+      spaAjaxEvents[interaction.id].forEach(function (item) {
+        // move it from the spaAjaxEvents buffer to the ajaxEvents buffer for harvesting here
+        ajaxEvents.push(item)
+      })
+      delete spaAjaxEvents[interaction.id]
+    })
+
+    if (allAjaxIsEnabled()) setDenyList(getConfigurationValue(agentIdentifier, 'ajax.deny_list'))
+
+    register('xhr', storeXhr, undefined, this.ee)
+
+    if (allAjaxIsEnabled()) {
+      scheduler = new HarvestScheduler('events', {
+        onFinished: onEventsHarvestFinished,
+        getPayload: prepareHarvest
+      }, this)
+
+      externalFeatures.forEach(feat => {
```

Comment:
This is the metrics harvest, should live outside of allAjaxIsEnabled check

Reply:
This is a nice catch.  This could be a new ticket for after the release, to extend the metrics feature to do more than just supportability and remove the dependency from AJAX feature.